lib/Gateway.js

```diff
@@ -1680,65 +1702,76 @@ Gateway.prototype.discoverValue = function (node, vId) {
+              )
               break
             case 'Door state':
-              cfg.discovery_payload.device_class =
+              cfg = setBinarySensorPayload(
                 Constants.deviceClass.sensor_binary.DOOR
+              )
               off = 1 // Door closed on payload 1
               break
+            case 'Alarm status':
             case 'Dust in device status':
+            case 'Load error status':
             case 'Over-current status':
             case 'Over-load status':
             case 'Hardware status':
-              cfg.discovery_payload.device_class =
+              cfg = setBinarySensorPayload(
                 Constants.deviceClass.sensor_binary.PROBLEM
+              )
               break
             case 'Heat sensor status':
-              cfg.discovery_payload.device_class =
+              cfg = setBinarySensorPayload(
                 Constants.deviceClass.sensor_binary.HEAT
+              )
               break
             case 'Motion sensor status':
-              cfg.discovery_payload.device_class =
+              cfg = setBinarySensorPayload(
                 Constants.deviceClass.sensor_binary.MOTION
+              )
               break
-            case 'Alarm status':
-              cfg.discovery_payload.device_class =
-                Constants.deviceClass.sensor_binary.CONNECTIVITY
+            case 'Water Alarm':
+              cfg = setBinarySensorPayload(
+                Constants.deviceClass.sensor_binary.MOISTURE
+              )
               break
             // sensor status has multiple Properties. therefore is good to work
             // on property basis... user friendly
             case 'Sensor status':
               switch (valueId.propertyName) {
                 case 'Smoke Alarm':
-                  cfg.discovery_payload.device_class =
+                  cfg = setBinarySensorPayload(
                     Constants.deviceClass.sensor_binary.SMOKE
+                  )
                   break
                 case 'Water Alarm':
-                  cfg.discovery_payload.device_class =
+                  cfg = setBinarySensorPayload(
                     Constants.deviceClass.sensor_binary.MOISTURE
+                  )
                   break
                 default:
               }
-              cfg.object_id = valueId.propertyName
-              break
-            case 'Water Alarm':
-              cfg.discovery_payload.device_class =
-                Constants.deviceClass.sensor_binary.MOISTURE
+              discoveredObjectId = valueId.propertyName
               break
             default:
           }
+          // cfg not there?
+          cfg = cfg ? copy(cfg) : copy(hassCfg.binary_sensor)
+          // correct payload from true/false to numeric values
           cfg = setBinaryPayloadFromSensor(cfg, valueId, off)
+          // finally update object_id
+          cfg.object_id = discoveredObjectId
```

Comment:
@billiaz Shouldn't this be:

```suggestion
          cfg.object_id += discoveredObjectId
```

Reply:
we replace the object ID from the default. to fit the one we need to use

Reply:
i agree, let me make some changes, as this  'binary_sensor_' prefix misses everywhere